h/static/scripts/annotator/guest.coffee

```diff
@@ -211,21 +161,131 @@ module.exports = class Guest extends Annotator
 
     this.removeEvents()
 
-  setupAnnotation: ->
-    annotation = super
-    this.plugins.CrossFrame.sync([annotation])
-    annotation
+  setupAnnotation: (annotation) ->
+    self = this
+    root = @element[0]
+
+    anchors = []
+    anchoredTargets = []
+    deadHighlights = []
+
+    annotation.target ?= []
+
+    locate = (target) ->
+      options = {
+        cache: self.anchoringCache
+        ignoreSelector: '[class^="annotator-"]'
+      }
+      return self.anchoring.anchor(root, target.selector, options)
+      .then((range) -> {annotation, target, range})
+      .catch(-> {annotation, target})
+
+    highlight = (anchor) ->
+      return anchor unless anchor.range?
+      return animationPromise ->
+        range = Annotator.Range.sniff(anchor.range)
+        normedRange = range.normalize(root)
+
+        highlights = highlighter.highlightRange(normedRange)
+        rect = highlighter.getBoundingClientRect(highlights)
+
+        $(highlights).data('annotation', anchor.annotation)
+
+        anchor.highlights = highlights
+        anchor.pos =
+          left: rect.left + window.scrollX
+          top: rect.top + window.scrollY
+
+        return anchor
+
+    sync = (anchors) ->
+      annotation.$anchors = ({pos} for {pos} in anchors)
+      annotation.$orphan = anchors.length > 0
+      for anchor in anchors
+        if anchor.range?
+          annotation.$orphan = false
+
+      self.anchors = self.anchors.concat(anchors)
+      self.plugins.BucketBar?.update()
+      self.plugins.CrossFrame?.sync([annotation])
+
+    for anchor in self.anchors.splice(0, self.anchors.length)
```

Comment:
I think Array.prototype.filter is supported for IE9 and up, so it might be clearer to use that here.